src/modm/board/nucleo_l031k6/board.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2016-2018, Niklas Hauser
+ * Copyright (c) 2017, Nick Sarten
+ * Copyright (c) 2017, Sascha Schade
+ * Copyright (c) 2021, Christopher Durand
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef MODM_STM32_NUCLEO_L031K6_HPP
+#define MODM_STM32_NUCLEO_L031K6_HPP
+
+#include <modm/platform.hpp>
+#include <modm/architecture/interface/clock.hpp>
+#include <modm/debug/logger.hpp>
+/// @ingroup modm_board_nucleo_l031k6
+#define MODM_BOARD_HAS_LOGGER
+
+using namespace modm::platform;
+
+/// @ingroup modm_board_nucleo_l031k6
+namespace Board
+{
+using namespace modm::literals;
+
+/// STM32L031K6 running at 32MHz generated from 16 MHz HSI16 clock
+struct SystemClock
+{
+	static constexpr uint32_t Frequency = 32_MHz;
+	static constexpr uint32_t Ahb  = Frequency;
+	static constexpr uint32_t Apb1 = Frequency;
+	static constexpr uint32_t Apb2 = Frequency;
+
+	static constexpr uint32_t Apb1Timer = Apb1;
+	static constexpr uint32_t Apb2Timer = Apb2;
+
+	static constexpr uint32_t Adc    = Apb2;
+
+	static constexpr uint32_t Spi1   = Apb2;
+
+	static constexpr uint32_t Usart2 = Apb1;
+
+	static constexpr uint32_t I2c1   = Apb1;
+
+	static constexpr uint32_t Timer2  = Apb1Timer;
+	static constexpr uint32_t Timer21 = Apb2Timer;
+	static constexpr uint32_t Timer22 = Apb2Timer;
```

Comment:
Most BSP have this entries alphabetically ordered.

You could add the Comp1 and Comp2 entries (= Apb2) if someone is going to enable/port the comparator driver for L0 in the future...

![grafik](https://user-images.githubusercontent.com/2820734/113358414-0bce5f00-9346-11eb-86fc-4d60b7305bfd.png)

Reply:
Thanks, now alphabetically ordered and with comparator clock.